transport/proxy.go

```diff
@@ -107,7 +108,11 @@ func doHTTPConnectHandshake(ctx context.Context, conn net.Conn, addr string) (_
 	}
 	resp.Body.Close()
 	if resp.StatusCode != http.StatusOK {
-		return nil, fmt.Errorf("failed to do connect handshake, status code: %s", resp.Status)
+		dump, err := httputil.DumpResponse(resp, true)
```

Comment:
I tested it in the proxy_test but forgot to set the response body...
Changed the close() to defer, and it works now.